Misc/NEWS.d/next/Library/2023-03-22-19-41-23.gh-issue-102821.nAt1VY.rst

```diff
@@ -0,0 +1 @@
+Simplify redundant logic of :meth:`threading.Event.wait`.
```

Comment:
This news entry isn't informative about the impact this patch will have on users of `threading` (the performance boost). It only tells me about the changes to the internal implementation, which users of the `threading` module likely won't care about.